assets/src/edit-story/app/story/effects/useLoadFonts.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useAPI } from '../../';
+
+function useLoadFonts( { fonts, setFonts } ) {
+	const { actions: { getAllFonts } } = useAPI();
+
+	useEffect( () => {
+		if ( fonts.length === 0 ) {
+			getAllFonts( {} ).then( ( loadedFont ) => {
```

Comment:
These three lines could simply be:
```suggestion
			getAllFonts( {} ).then( setFonts );
```